### PR TITLE
api/types/client.go: documentation fix for ImageImportSource

### LINKED ISSUE #27922: Docker server implementation of image import API does not respect request body in some scenarios
**Description**

The docker engine server-side API does not respect the Source parameter provided by the API (which specifies a reader that can be used). When specifying a Source with no SourceName (they are described as mutually exclusive in the docs), it tries to process an empty string as a URL and while doing that successfully producing `http://`, yields no error except for in the response body, which makes it really fun to catch. :)

**Steps to reproduce the issue:**
1. use the docker api for image import with no SourceName and an open io.Reader attached to Source
2. monitor err and response reader, the latter will be readable with json output stating the error. the former will be nil.
3. observe it is trying to pull `http://`

**Additional information you deem important (e.g. issue happens only occasionally):**

I think the relevant code is here: look at this https://github.com/docker/docker/blob/master/api/server/router/image/image_routes.go#L109-L115

I don't believe the body is considered at all without examining the post parameters first.

Is this a case of wrong docs?

https://godoc.org/github.com/docker/docker/api/types#ImageImportSource

**Output of `docker version`:**

```
Client:
 Version:      1.12.3
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   6b644ec
 Built:        Wed Oct 26 22:01:48 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.3
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   6b644ec
 Built:        Wed Oct 26 22:01:48 2016
 OS/Arch:      linux/amd64

```

**Output of `docker info`:**

not using the docker client but I'm using the 1.12.3 tag and also tried docker/engine-api before finding out it was deprecated.

**Additional environment details (AWS, VirtualBox, physical, etc.):**

building with go 1.7.3, but i really doubt that matters

### PR DESCRIPTION
**Pull request by erikh**
**- What I did**

Fixes #27922 

**- Description for the changelog**

Fixes a documentation issue with types.ImageImportSource.

Thanks!

### CODE REVIEW

File: api/types/client.go [hunk 1]
```diff
@@ -173,8 +173,8 @@ type ImageCreateOptions struct {
 
 // ImageImportSource holds source information for ImageImport
 type ImageImportSource struct {
-	Source     io.Reader // Source is the data to send to the server to create this image from (mutually exclusive with SourceName)
-	SourceName string    // SourceName is the name of the image to pull (mutually exclusive with Source)
+	Source     io.Reader // Source is the data to send to the server to create this image from. Must SourceName to "-" to leverage this.
```
Comment (thaJeztah) [inline api/types/client.go]: "Must SourceName to"

Incomplete sentence? :innocent:

**Comment by erikh** [comment]
bah that's what I get for being lazy

first patch to docker in 2 years; doesn't compile

**Comment by erikh** [comment]
anyway it should be better now

**Review by thaJeztah — APPROVED** [review]
LGTM pending https://github.com/docker/docker/issues/27922#issuecomment-257486981

**Review by vdemeester — APPROVED** [review]
LGTM 🐸

**Pull request merged**